crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -268,22 +272,8 @@ impl From<&Account> for AccountInterface {
         let components = AccountComponentInterface::from_procedures(account.code().procedures());
         let mut auth = Vec::new();
         components.iter().for_each(|interface| {
-            match interface {
-                // RpoFalcon512, RpoFalcon512Acl, and RpoFalconMultisig use the same RpoFalcon512
-                // auth scheme
-                AccountComponentInterface::AuthRpoFalcon512(storage_index)
-                | AccountComponentInterface::AuthRpoFalcon512Acl(storage_index)
-                | AccountComponentInterface::AuthRpoFalconMultisig(storage_index) => {
-                    auth.push(AuthScheme::RpoFalcon512 {
-                        pub_key: rpo_falcon512::PublicKey::new(
-                            account
-                                .storage()
-                                .get_item(*storage_index)
-                                .expect("invalid storage index of the public key"),
-                        ),
-                    })
-                },
-                _ => {},
+            if let Some(auth_scheme) = interface.get_auth_scheme(account.storage()) {
+                auth.push(auth_scheme);
```

Comment:
Related to the previous couple of comments: I don't think this code is technically correct. Specifically:

- An account should have only one auth component. So, once we found it, we don't need to continue iterating over components.
- We should be able to get multiple schemes from the auth component.

So, this could look something like:
```rust
for component in components.iter() {
    if component.is_auth_component() {
        auth = component.get_auth_schemes();
        break;
    }
}
```